src/main/java/seedu/address/model/student/Email.java

```diff
@@ -35,8 +35,10 @@
      * @param email A valid email address.
      */
     public Email(String email) {
-        requireNonNull(email);
-        checkArgument(isValidEmail(email), MESSAGE_CONSTRAINTS);
+        if (!email.equals("")) {
+            requireNonNull(email);
```

Comment:
Require nonNull should be outside. Note that "" is not the same as null.
Alternative is -> if (!"".equals(email)). Currently passing in a null will throw an exception.